pygmt/src/coast.py

```diff
@@ -183,10 +188,22 @@ def coast(self, **kwargs):
     {V}
     """
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
+
+    if "clip" in kwargs and kwargs["clip"] == "land":
+        kwargs["G"] = True
+        kwargs.pop("clip")
+    elif "clip" in kwargs and kwargs["clip"] == "water":
+        kwargs["S"] = True
+        kwargs.pop("clip")
+    elif "clip" in kwargs and kwargs["clip"] == "end":
+        kwargs["Q"] = True
+        kwargs.pop("clip")
```

Comment:
No need to check `"clip" in kwargs` in three if-branches:
```suggestion
    if "clip" in kwargs:
        if kwargs["clip"] == "land":
            kwargs["G"] = True
            kwargs.pop("clip")
        elif kwargs["clip"] == "water":
            kwargs["S"] = True
            kwargs.pop("clip")
        elif kwargs["clip"] == "end":
            kwargs["Q"] = True
            kwargs.pop("clip")
```